### PR TITLE
Cleanup typedefs, add QoL tx functionality, fix UTXO spending bug

### PR DESCRIPTION
**Pull request by ACatThatPrograms**
_**Covers the following**_

**Adds wait-able sendTx function function similar to ethers**

So this is now possible:
```
let tx = await wallet.sendWaitableTx();
let hash = tx.txHash;
let receipt = await tx.wait();
```

**Fixes UTXO Spending Bug**
- Calculation fixed on UTXO spending logic

** Fixes typedefs **
- Instead of importing module, import type

### CODE REVIEW

File: src/Transaction.js [hunk 1]
```diff
@@ -1,28 +1,79 @@
 const Tx = require('./Transaction/Tx.js');
 const Constants = require('./Constants.js');
+
 /**
  * Transaction handler
- * @class Transaction
+ * @class
+ * @property {import('./Wallet.js')} Wallet - Circulare Wallet reference
+ * @property {Tx} Tx - The tranasaction object to be sent
```
Comment (everton0991) [inline src/Transaction.js:8]: Spelling - Transaction

File: src/Transaction.js [hunk 2]
```diff
@@ -50,6 +101,38 @@ class Transaction {
...
+            }
+            if (this.Tx.Vout.length <= 0) {
+                throw "No Vouts for transaction"
+            }
+            if (!this.Wallet.Rpc.rpcServer) {
+                throw 'No RPC to send transaction'
+            }
+            await this._createTxIns(changeAddress, changeAddressCurve, UTXOIDs);
+            await this.Tx._createTx();
+            console.log("\nSENDING: \n", JSON.stringify(this.Tx.getTx(), false, 2));
```
Comment (everton0991) [inline src/Transaction.js:125]: Do we need this console log in prod?

File: src/Transaction.js [hunk 3]
```diff
@@ -425,10 +509,13 @@ class Transaction {
                         if (reward) {
                             await this._createDataTxIn(account["address"], DS);
                             outValue["totalValue"] = BigInt(outValue["totalValue"]) - BigInt(reward);
+                        } else {
+                            // No reward
                         }
+                    } else {
+                        // Skip if the store doesn't equal the datastore for spending
```
Comment (everton0991) [inline src/Transaction.js:516]: Same as prev comment

File: src/Transaction.js (hunk 2 above)
Comment (ACatThatPrograms) [reply src/Transaction.js:125]: No, good catch thank you. leftover from the bug fix in the other PR haha :+1:

File: src/Transaction.js (hunk 3 above)
Comment (ACatThatPrograms) [reply src/Transaction.js:516]: Will clean these up, thank you.

**Pull request merged**